bson/bsoncodec/time_codec.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncodec
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/bsonoptions"
+	"go.mongodb.org/mongo-driver/bson/bsonrw"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+)
+
+const (
+	timeFormatString = "2006-01-02T15:04:05.999Z07:00"
+)
+
+var defaultTimeCodec = &TimeCodec{}
+
+// TimeCodec is the Codec used for struct values.
+type TimeCodec struct {
+	UseLocalTimeZone bool
+	DecodeFromString bool
+}
+
+var _ ValueEncoder = &TimeCodec{}
```

Comment:
nit: You could assert that this implements `ValueCodec`, which combines `ValueEncoder` and `ValueDecoder` into a single interface.

Reply:
Done